heron/common/src/java/com/twitter/heron/common/utils/logging/ErrorReportLoggingHandler.java

```diff
@@ -75,13 +85,20 @@ public void publish(LogRecord record) {
     // Convert Log
     Throwable throwable = record.getThrown();
     if (throwable != null) {
-      StringWriter sink = new StringWriter();
-      throwable.printStackTrace(new PrintWriter(sink, true));
-      String trace = sink.toString();
       synchronized (ExceptionRepositoryAsMetrics.INSTANCE) {
+        // We would not include the message if already exceeded the exceptions limit
+        if (ExceptionRepositoryAsMetrics.INSTANCE.getExceptionsCount() >= exceptionsLimit) {
```

Comment:
What's the scope of this counter? Does this mean that no more exceptions will ever be published, or just for this payload? Same for `droppedExceptionsCount`. Is that count per message payload, count since the jvm started, or what?